inc/classes/subscriber/Optimization/class-lazyload-subscriber.php

```diff
@@ -426,6 +427,22 @@ public function deactivate_lazyload_on_specific_posts() {
 		}
 	}
 
+	/**
+	 * Disable WP core lazyload if our images lazyload is active
+	 *
+	 * @since 3.5
+	 * @author Remy Perona
+	 *
+	 * @return bool
+	 */
+	public function maybe_disable_core_lazyload() {
```

Comment:
I think this method name is misleading. Why? We are returning `false` to disable it and `true` to enable it. When I first read this method name, I thought the inverse. To keep it crystal clear, I'd suggest changing the name to:

```php
public function maybe_enable_wp_native_lazyload() {
```

Think about it. If `true`, it enables WP native lazyload. That happens when ours cannot lazyload images, i.e. when `can_lazyload_images() === true`.

Reply:
The native lazyload will be enabled by default in core (default value for the filter is true). That's why I felt naming the method like this made more sense: we want to disable core lazyload when our lazyload is active

Reply:
Let's walk through what the code is doing and see if the callback _tells_ us the expected behavior:

- WordPress fires a filter called `'wp_lazy_loading_enabled'` and passes a `true` value to turn on and enable this feature.
- Our callback is fired.
- What does it mean when our callback returns `true`? Does `true` === "disable"? No, it means "enable" native core.

By convention, we think about a `true` state as doing the what the method tells us its behavior is and what will happen when it's invoked. So in this case, the expected behavior is not to disable, but to enable, where a `true` being returned means "enable" it.

Reply:
I updated the method a bit, to take into account the possibility that something else changed the default value before us (we're hooking on the default priority 10, so that's possible).

Instead of returning true/false, I'm returning false if our lazyload is active as that's the behavior we need, but if it's not, I'm returning the value passed to the callback, which can be eithere true or false, but we don't have to care about it.

This also makes the method name more appropriate, since now we disable if we need, but do nothing if we don't.

What do you think?

Reply:
>to take into account the possibility that something else changed the default value before us (we're hooking on the default priority 10, so that's possible).

@Tabrisrp That's a very good point. What if it's already set to `false`? Then we don't need to check if ours is enabled or not. Right? If the given value is `false`, bail out and return it. Else, check if ours is enabled.